File: 2816-lexicographically-smallest-palindrome/2816-lexicographically-smallest-palindrome.py
class Solution(object):
    def makeSmallestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        left=0
        right=len(s)-1
        s=list(s)
        while(left<=right):
            if(s[left]>s[right]):
                s[left]=s[right]
            elif(s[right]>s[left]):
                s[right]=s[left]
            left+=1
            right-=1
        return ''.join(s)


        

File: 2816-lexicographically-smallest-palindrome/2816-lexicographically-smallest-palindrome.java
class Solution {
    public String makeSmallestPalindrome(String s) {
        int left=0;
        int right=s.length()-1;
        StringBuilder sb=new StringBuilder(s);
        while(left<=right){
            if(sb.charAt(left)>sb.charAt(right)){
                sb.setCharAt(left,sb.charAt(right));
            
                
            }
            else if(sb.charAt(left)<sb.charAt(right)){
                sb.setCharAt(right,sb.charAt(left));
            }
            left++;
            right--;
        }
        return sb.toString();
        
    }
}